### PR TITLE
Changes in setup

### PR DESCRIPTION
**Pull request by sonalkr132**
Users will have different config for postgres (username, password etc).
It makes more sense to given them an example file which they use to create
their own config/database.yml. They won't have to worry about git tracking
the changes.

redis should start before we run any migration.
rake db:setup creates database too, there is no need of db:create.

### CODE REVIEW

File: config/database.yml.example [hunk 1]
```diff
@@ -14,12 +14,15 @@
 # Configure Using Gemfile
 # gem 'pg'
 #
+# For details on connection pooling, see rails configuration guide
+# http://guides.rubyonrails.org/configuring.html#database-pooling
```
Comment (tgxworld) [inline config/database.yml.example]: Any reason this needs to be moved up?

**Comment by tgxworld** [comment]
@sonalkr132  Thanks for the PR :+1:  Just one minor comment.

File: config/database.yml.example (hunk 1 above)
Comment (sonalkr132) [inline config/database.yml.example]: That link doesn't really give much info about setting up pg database.

Comment (tgxworld) [inline config/database.yml.example]: It provide details about connection pooling which I think should stay where it is. Plus it is the default anyway.

**Comment by sonalkr132** [comment]
@tgxworld I have updated my PR

**Comment by tgxworld** [comment]
@sonalkr132  Thanks!!! I'm sorting out the broken tests now.

**Pull request merged**